### DISCUSSION TITLE
Support for "Update" integration in Home Assistant

### DISCUSSION THREAD

**Discussion by diegofrances** [arendst/Tasmota#21950] [Ideas]
Home Assistant has developed a new feature to show if there is any update available for a device or service. This can be any update, including update of a firmware for a device like a light bulb or router, or software updates for things like add-ons or containers.

Please find some details below:

https://www.home-assistant.io/integrations/update/

Currently Tasmota is not included on the list of integrations offering update entities (https://www.home-assistant.io/integrations/#update).

I truly believe that this feature will be really appreciated and loved by all Tasmota's users as there is no way to know if there is any firmware update unless you visit the Tasmota site every once in a while.

Thank you in advance for taking this request in consideration.

Best Regards.

**Comment by sfromis** [comment]
Was also discussed a bit on Discord:
https://discord.com/channels/479389167382691863/790182550085304342/1272561185228591124

**Comment by stefanbode** [comment]
Updating is an important but difficult process. See Crowdstrike. Anyhow I would also love to replace my homegrown http update process with something ootb. Not all devices are eligible for updates.  Here I removed the update URL.

**Comment by barbudor** [comment]
I don't think this needs anything from Tasmota firmware. It's not to the device to search for new firmware.
If you have 100 devices, why would you want every device to scan perdiodically for new firmware, especially if you don't want them to access the Internet ?

The repository already includes a version file that can be polled by HATasmota integration to know about latest release and development version. They HA can have a menu to send an OTA upgrade command to every or selected devices.

This is of course totally useless for people using custom build firmware such as myself.
I manage my upgrades through TDM.

**Comment by arendst** [comment]
I agree with @barbudor and won't add any updated version related advertising to Tasmota binaries.

**Reply by diegofrances** [reply]
Hello @arendst and @barbudor,

Maybe my request has been misunderstood or I have not been able to understand the HA feature and I have led you to a wrong understanding of my request.

I thought that in order for HA to provide this feature, they should work with Tasmota devices somehow so I thought that a new feature or collaboration from Tasmota Team was required.

I totally agree with you @barbudor regarding that 99.9% of the people that installs Tasmota on their devices is because does not want them to leave their Network or go to any cloud or similar so I apologize if I have led you to a wrong idea.

Based on all the information you have shared I believe that the request should be posted on HA Forum rather than on Tasmota one.

Again, I apologize and I appreciate your time on reading my request.

**Comment by FaserF** [comment]
Pull request is submitted which would implement this
https://github.com/emontnemery/hatasmota/pull/389

For reference: 
https://github.com/arendst/Tasmota/discussions/15480
https://github.com/emontnemery/hatasmota/issues/151
